### PR TITLE
User will now get a point for news/stories interactions

### LINKED ISSUE #156: Update point system to include input from Camper News

### PR DESCRIPTION
**Pull request by terakilobyte**
User will get one point for a story submission as well as a point for every upvote on a submitted story. Closes #156

### CODE REVIEW

File: controllers/story.js [hunk 1]
```diff
@@ -1,431 +1,449 @@
...
-            return next(err);
-        }
-        res.json(stories);
-    });
+  var story = Story.find({}).sort({'timePosted': -1}).limit(100);
+  story.exec(function(err, stories) {
+    if (err) {
+      res.status(500);
+      return next(err);
+    }
```
Comment (BerkeleyTrue) [inline controllers/story.js]: You should remove the `next` call here and call `res.end()`.

File: controllers/story.js [hunk 2]
```diff
@@ -1,431 +1,449 @@
...
-    var storyName = dashedName.replace(/\-/g, ' ');
+  var storyName = dashedName.replace(/\-/g, ' ');
 
-    Story.find({'storyLink' : new RegExp(storyName, 'i')}, function(err, story) {
-        if (err) {
-            next(err);
-        }
+  Story.find({'storyLink': new RegExp(storyName, 'i')}, function(err, story) {
+    if (err) {
+      next(err);
```
Comment (BerkeleyTrue) [inline controllers/story.js]: Need a return here to return early.

File: controllers/story.js [hunk 3]
```diff
@@ -1,431 +1,449 @@
...
-            }
-        }).toArray(function(err, items) {
-            if (items !== null && items.length !== 0) {
-                return res.json(items);
-            }
-            return res.status(404);
-        });
+  MongoClient.connect(secrets.db, function(err, database) {
+    if (err) {
+      return res.status(500);
```
Comment (BerkeleyTrue) [inline controllers/story.js]: need a call to `res.end` as well or the request will never be ended. You could chain this as well `res.status(500).end();`

File: controllers/story.js [hunk 4]
```diff
@@ -1,431 +1,449 @@
...
+      }
+    }, {
+      sort: {
+        textScore: {
+          $meta: 'textScore'
+        }
+      }
+    }).toArray(function(err, items) {
+      if (err) {
+        return res.status(500);
```
Comment (BerkeleyTrue) [inline controllers/story.js]: same as above.

File: controllers/story.js [hunk 5]
```diff
@@ -1,431 +1,449 @@
...
+    story.save();
+    User.find({'_id': story.author.userId}, function(err, user) {
+      'use strict';
+      if (err) {
+        return next(err);
+      }
+      user = user.pop();
+      user.progressTimestamps.push(Date.now());
+      user.save();
     });
```
Comment (BerkeleyTrue) [inline controllers/story.js]: There are a couple of async calls here that need to be nested or promisify and chained here.

**Comment by BerkeleyTrue** [comment]
@terakilobyte there are a bunch of cases here where the error is being handled here by responding with status 500, but if you pass the error to next the status is automatically set to 500 by the error handler middleware. To be consistent these should be changed to `return next(err);`

**Comment by terakilobyte** [comment]
Easy enough. No longer sending 500 to user on mongo errors. Also, poking through the documentation as well as reading elsewhere, sendStatus is the future-proof method as referenced [here](https://github.com/strongloop/express/issues/2269)

**Comment by BerkeleyTrue** [comment]
Looks like you are still sending 500 down in a couple spots. All errors should be handled by https://github.com/FreeCodeCamp/freecodecamp/blob/master/app.js#L537

**Comment by terakilobyte** [comment]
I was under the impression from the documentation for this errorhandler that it is for development purposes only.

**Comment by BerkeleyTrue** [comment]
I have added a production error handler now so that should not be an issue. But yes it was for use only during development but it was not being turned off.

**Pull request merged**